### PR TITLE
Site Badge: apply colors by context

### LINKED ISSUE #49180: Navigation colors bleeding to "coming soon" badges at Notifications
At https://wordpress.com/me/notifications I see dark colored badges:

Before

Screenshot 2021-01-22 at 12 48 15

After

Screenshot 2021-01-22 at 12 48 31

### PR DESCRIPTION
**Pull request by retrofox**
#### Changes proposed in this Pull Request

This PR applies the site badge colors by context, meaning that sometimes it will apply the colors defined by the nav unification (when the badge is rendered in the secondary layout / sidebar), of applying neutral colors in other contexts, for instance, in the primary layout.

Fixes https://github.com/Automattic/wp-calypso/issues/49180

#### Testing instructions

* Confirm the visual issue described here https://github.com/Automattic/wp-calypso/issues/49180
Basically, you should see the colors defined by the nav unification.

* applying the changes on this branch, you should see the badge with neutral colors when it's rendered in a different context than the secondary layout, for instance, testing in the notifications page:

Screen Shot 2021-03-29 at 12 45 55 PM

* In the secondary layout, the colors should be defined by the nav unification.

Screen Shot 2021-03-29 at 12 48 02 PM

Related to https://github.com/Automattic/wp-calypso/issues/49180

### CODE REVIEW

**Comment by pablinos** [comment]
Looks good. I saw that[ it was mentioned that the site selector](https://github.com/Automattic/wp-calypso/issues/49180#issuecomment-780423053) was also suffering from this problem. I checked it with this branch, and I think it may have made the hover state worse for my colour scheme:

image

I'm not sure what it looked like before the nav unification though, and using the `disable-nav-unification` parameter has no effect.

**Comment by retrofox** [comment]
> Looks good. I saw that[ it was mentioned that the site selector](https://github.com/Automattic/wp-calypso/issues/49180#issuecomment-780423053) was also suffering from this problem. I checked it with this branch, and I think it may have made the hover state worse for my colour scheme:

I checked this issue too, and I thought it's already fixed. What color scheme are you using for testing? I've tested with several ones and I couldn't reproduce the visual issue.

**Comment by pablinos** [comment]
The one in that screenshot was Midnight. I've just tried it with Ectoplasm and you get a similar effect. 

image

I'm not sure if we're looking to just make the white text on the badge readable or to neutralise all the colours on the hover state. The production version looks like this:

image

**Comment by retrofox** [comment]
Sorry I got it. My bad. Working on this now...

**Comment by retrofox** [comment]
> I'm not sure if we're looking to just make the white text on the badge readable or to neutralise all the colours on the hover state. The production version looks like this:

I'd suggest keeping the things like production for now. I [asked for design feedback](https://github.com/Automattic/wp-calypso/issues/49180#issuecomment-809595955) on the original issue.

**Comment by pablinos** [comment]
> I'd suggest keeping the things like production for now

Yeah makes sense. Let's treat it as a two-part solution

**Review by pablinos — APPROVED** [review]
Looks good. Let's follow up with the site selector fix.
